bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Component.java

```diff
@@ -50,12 +56,14 @@ default String getId() {
     /**
      * Returns the data layer information associated with the component
      *
-     * @return {@link DataLayer} object associated with the component
+     * @return {@link ComponentDataModel} object associated with the component
      *
      * @since com.adobe.cq.wcm.core.components.models 12.14.0
      */
     @Nullable
-    default DataLayer getDataLayer() {
+    @JsonProperty("dataLayer")
+    @JsonSerialize(using = ComponentDataModelSerializer.class)
+    default ComponentDataModel getComponentDataModel() {
         throw new UnsupportedOperationException();
```

Comment:
To avoid needing to bump up versions for all component we could replace this with a default of `return null`.

Reply:
Added `return null` in 5dbfd4e